util/opentelemetry-util-http/src/opentelemetry/util/http/asgi/__init__.py

```diff
@@ -27,10 +26,10 @@
 from asgiref.compatibility import guarantee_single_callable
 
 from opentelemetry import context, propagators, trace
-from opentelemetry.instrumentation.asgi.version import __version__  # noqa
 from opentelemetry.instrumentation.utils import http_status_to_status_code
 from opentelemetry.trace.propagation.textmap import DictGetter
 from opentelemetry.trace.status import Status, StatusCode
+from opentelemetry.util.http.asgi.version import __version__  # noqa
```

Comment:
Not sure about moving the WSGI/ASGI `OpenTelemetryMiddleware` classes to the `opentelemetry-util-http` package. With this there are now 3 `version.py` files in the  package since both `OpenTelemetryMiddleware` implementations are de facto instrumentations which require the version to get the tracer for producing spans.


Imo WSGI/ASGI `OpenTelemetryMiddleware` should stay in their own dedicated packages (common helper methods for extracting span attriubtes, ... can be moved to the util package though), also for dependency reasons.  E.g. if one wants instrument an arbitrary WSGI application with the WSGI `OpenTelemetryMiddleware` from the `opentelemetry-util-http` the `asgiref` library will be pulled additionally even though it is not required.

Reply:
@mariojonke @lonewolf3739 

I understand that the current ASGI and WSGI packages are closely related to instrumentation. Nevertheless, they are not complete instrumentations as the rest of the instrumentations because they are not instrumenting anything by themselves when auto instrumentation happens nor they implement `BaseInstrumentor`. For that reason I believe they should not be part of any `instrumentation` package. The idea of `util` is pretty much to have a place where we can put anything that does not fit in the category of exporter, instrumentation, api, sdk, etc. I believe this to be the case, even when I see how this approach causes some concern because of the close relationship between the ASGI and WSGI and the instrumentation of other third party libraries.

Maybe having a `util.instrumentation.asgi`/`wsgi` package would be a better solution? :shrug: